orcid-commons/src/main/java/no/sikt/nva/orcid/commons/service/OrcidServiceImpl.java

```
@@ -1,11 +1,57 @@
+        this.serviceWithTransactions = new ServiceWithTransactions(client, orcidTableName);
+        this.clockForTimestamps = clock;
+        this.readOrcidCredentialsService = new ReadOrcidCredentialsService(client, orcidTableName);
+    }
+
+    @Override
+    public OrcidCredentials createOrcidCredentials(OrcidCredentials orcidCredentials) {
+        var currentTime = clockForTimestamps.instant();
+        orcidCredentials.setModified(currentTime);
+        orcidCredentials.setCreated(currentTime);
+        return insertOrcidCredentials(orcidCredentials);
+    }
+
     @Override
-    public OrcidCredentials insertOrcidCredentials(OrcidCredentials orcidCredentials) {
-        return null;
+    public OrcidCredentials fetchOrcidCredentialsByOrcid(URI orcid) {
+        var orcidCredentials = new OrcidCredentials();
+        orcidCredentials.setOrcid(orcid);
+        return readOrcidCredentialsService.getOrcidCredentials(orcidCredentials);
+    }
+
+    private OrcidCredentials insertOrcidCredentials(OrcidCredentials orcidCredentials) {
+        TransactWriteItem transactionItem = transactionItemsForOrcidCredentialsInsertion(orcidCredentials);
+        TransactWriteItemsRequest putRequest = newTransactWriteItemsRequest(transactionItem);
+        serviceWithTransactions.sendTransactionWriteRequest(putRequest);
+        return fetchSavedOrcidCredentials(orcidCredentials);
+    }
+
+    private TransactWriteItem transactionItemsForOrcidCredentialsInsertion(OrcidCredentials orcidCredentials) {
+        return
+            serviceWithTransactions.newPutTransactionItem(new OrcidCredentialsDao(orcidCredentials));
+    }
+
+    private OrcidCredentials fetchSavedOrcidCredentials(OrcidCredentials orcidCredentials) {
+        return serviceWithTransactions
+                   .fetchEventualConsistentDataEntry(orcidCredentials,
+                                                     readOrcidCredentialsService::getOrcidCredentials)
+                   .orElse(null);
```

Probably just return the optional and delegate the decision about what to do with this to the caller

---

On second thought: I think I want to throw an exception. Having public Optional<orcidCredentials> createOrcidCredentials in orcidService will make it difficult to assert what went wrong.